runtime/src/main/java/org/corfudb/runtime/clients/BaseSenderClient.java

```diff
@@ -0,0 +1,102 @@
+package org.corfudb.runtime.clients;
+
+import java.util.concurrent.CompletableFuture;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.protocols.wireprotocol.CorfuMsg;
+import org.corfudb.protocols.wireprotocol.CorfuMsgType;
+import org.corfudb.protocols.wireprotocol.CorfuPayloadMsg;
+import org.corfudb.protocols.wireprotocol.VersionInfo;
+
+/**
+ * This is a base client which processes basic messages.
+ * It mainly handles PINGs, as well as the ACK/NACKs defined by
+ * the Corfu protocol.
+ *
+ * <p>Created by zlokhandwala on 2/20/18.
+ */
+@Slf4j
+public class BaseSenderClient implements IClient {
```

Comment:
Why does it inherit from IClient ? It does not override of use any of the methods  of the IClient?

Reply:
I kept it the same so as not to break anything else as this is considered a client.
If this is the correct approach, moving forward we should split this into 2 interfaces IHandlerClient and ISenderClient